src/main/java/org/cirdles/topsoil/app/TSVTable.java

```diff
@@ -41,8 +41,8 @@
  */
 public class TSVTable extends TableView<Entry> {
 
-    private Path savePath;
-    private Dataset dataset;
+    private Optional<DatasetResource> datasetResource = Optional.empty();
+    private Optional<Dataset> unsavedDataset = Optional.empty(); //Due to pasteFromClipboard
```

Comment:
Why can't the data from the clipboard be its own `DatasetResource`? Then you have `FileDatasetResource` and `ClipboardDatasetResource`.